File: backend/src/main/java/com/inventory/backend/controller/CartridgeModelController.java
package com.inventory.backend.controller;

import com.inventory.backend.entity.CartridgeModel;
import com.inventory.backend.service.CartridgeModelService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/cartridge-models")
@RequiredArgsConstructor
@CrossOrigin
public class CartridgeModelController {

    private final CartridgeModelService cartridgeModelService;

    @GetMapping
    public List<CartridgeModel> getAll() {
        return cartridgeModelService.getAll();
    }

    @PostMapping
    public CartridgeModel create(@Valid @RequestBody CartridgeModel cartridgeModel) {
        return cartridgeModelService.create(cartridgeModel);
    }

    @DeleteMapping("/{id}")
    public void delete(@PathVariable Long id) {
        cartridgeModelService.delete(id);
    }
}

File: backend/src/main/kotlin/com/inventory/backend/controller/CartridgeModelController.kt
package com.inventory.backend.controller

import com.inventory.backend.entity.CartridgeModel
import com.inventory.backend.service.CartridgeModelService
import jakarta.validation.Valid
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.CrossOrigin
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/cartridge-models")
@CrossOrigin
class CartridgeModelController(
    private val cartridgeModelService: CartridgeModelService,
) {
    @GetMapping
    @PreAuthorize("@authz.canViewCatalog()")
    fun getAll(): List<CartridgeModel> = cartridgeModelService.getAll()

    @PostMapping
    @PreAuthorize("@authz.canEditCatalog()")
    fun create(@Valid @RequestBody cartridgeModel: CartridgeModel): CartridgeModel =
        cartridgeModelService.create(cartridgeModel)

    @PutMapping("/{id}")
    @PreAuthorize("@authz.canEditCatalog()")
    fun update(
        @PathVariable id: Long,
        @Valid @RequestBody cartridgeModel: CartridgeModel,
    ): CartridgeModel = cartridgeModelService.update(id, cartridgeModel)

    @DeleteMapping("/{id}")
    @PreAuthorize("@authz.canEditCatalog()")
    fun delete(@PathVariable id: Long) {
        cartridgeModelService.delete(id)
    }
}
